bcgov/PIMS
# PIMS-67/71 CRA to Vite & Replace http-proxy-middleware

## 🎯 Summary

  
[PIMS-67:](https://citz-imb.atlassian.net/browse/PIMS-67?atlOrigin=eyJpIjoiYWE1MmE5NWI4Mzg3NDY2Nzk0YTNiMGU4YmNlMTg2OGMiLCJwIjoiaiJ9) SPIKE: Switch CRA to Vite React App
[PIMS-71:](https://citz-imb.atlassian.net/browse/PIMS-71?atlOrigin=eyJpIjoiYTgzZDkyZDIwOTYzNDZmYWI5MzQ4YzFhNWU4YzA0NjgiLCJwIjoiaiJ9) Investigation Replace http-proxy-middleware


This PR finalizes the move away from Create React App, using Vite as the primary frontend framework.
Prior to this, the app could not be run locally in a non-containerized environment, and the dev container used CRA still.

### Changes
- Replaced commands in `package.json` to use Vite only. These commands changed:
  - `npm run dev`: Starts the frontend in local development
  - `npm run build`: Builds the app for deployment. Used in Dockerfile for prod.
  - `npm start`: Removed. From CRA. Didn't work.
  - `npm run build-ocp`: Removed. A command that would use Vite to build the app. This is now just `npm run build`.
- Convert `viteConfig.js` to `viteConfig.ts` and add proxy functionality to mirror existing proxy. Some environment variables can be set here, but defaults match our current local setup to make it easier for devs.
- Add `.env-template` to aid new devs in setting up the PIMS project. There is no need for a `.env` in the `/frontend` folder for running the frontend unless you need to specify the app or api port to something other than the default.
- Updated Dockerfiles to accommodate changes 
- Separate dev container in docker-compose file. *Devs should talk about if we still want this*
- Update Makefile to reflect changes
- Removed http-proxy-middleware and associated `setupProxy.js` file.

### Testing
1. Ensure .env in root folder has expected key/values
2. Make sure any existing frontend containers are stopped.
3. In `/frontend`, run `npm run dev` and check if frontend is working. Stop it afterwards with CTRL+c
4. In the root, test dev container with `docker-compose up frontend-dev --build -d`. Stop container afterwards.
5. In the root, test production container with `docker-compose up frontend --build -d`


### After Merge
- Check OC environment variables
- Update documentation

## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
